src/main/java/io/jenkins/plugins/eggplant/EggplantRunnerBuilder.java

```diff
@@ -400,4 +472,140 @@ public static boolean isValidPath(String path, String extension) {
    
     }
 
+    public static abstract class TestConfig implements ExtensionPoint, Describable<TestConfig> {
+
+        public Descriptor<TestConfig> getDescriptor() {
+            return Jenkins.get().getDescriptor(getClass());
+        }
+    }
+
+    public static class TestConfigDescriptor extends Descriptor<TestConfig> {
+    } 
+
+    public static class TestConfigId extends TestConfig {
+        private final String id;
+
+        @DataBoundConstructor 
+        public TestConfigId(String id) {
+            this.id=id;
+        }
+
+        public String getId() {
+            return id;
+        }
+
+        @Extension
+        public static final class DescriptorImpl extends TestConfigDescriptor {
```

Comment:
```suggestion
        public static final class DescriptorImpl extends Descriptor<TestConfig> {
```

I will suggest to direct use the Descriptor<TestConfig> , straight forward and clear cut. No need to maintain another class.

Reply:
Understood it's for code maintainability. It's updated.